slice-core/src/main/java/com/cognifide/slice/core/internal/monitoring/ModelUsageData.java

```diff
@@ -0,0 +1,136 @@
+/*-
+ * #%L
+ * Slice - Core
+ * %%
+ * Copyright (C) 2012 Cognifide Limited
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.slice.core.internal.monitoring;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class ModelUsageData implements Map<Class<?>, ModelUsageData> {

```

Comment:
We were supposed to introduce composition instead of inheritance here. But this looks even more complicated.
Consider:
1. Leaving the composition
2. Dropping 'implements Map....'
3. Exposing subModel through a **get** method